README

```diff
@@ -1133,7 +1133,8 @@ NETWORKING SUPPORT / OPTIONS
   Specify the directory where the OpenFabrics Interfaces libfabric
   library and header files are located.  This option is generally only
   necessary if the libfabric headers and libraries are not in default
-  compiler/linker search paths.
+  compiler/linker search paths. Note: Open MPI 4.1.6 or older requires
+  libfabric 1.x headers.
```

Comment:
Don't we have another place where we record things like this?  Also, the headers thing is being *WAY* too pedantic.  Just say that Open MPI 4.1.6 only supports Libfabric 1.x.

Reply:
Finally, this patch should be its own PR.

Reply:
Hmmm let me omit it from this PR.

I think `--with-libfabric` section is the most relevant since the problem so far is the ofi header file.

@jsquyres for 4.1.x do we need to touch docs other than README.

Reply:
The README is basically the only docs we have for v4.1.x -- that being said, I didn't check the old FAQ to see if we have any general statements about 3rd party library versions.  You might want to look there.

The sentiment that we want to convey is that Open MPI v4.1.6 and earlier will not compile successfully with libfabric 2.x.

Reply:
Good callout for FAQ. I will update the web page for both 4.1.x and 5.0.x.

Meanwhile, rephrase the help message here https://github.com/open-mpi/ompi/pull/12733

Reply:
Keep in mind that the old FAQ is really only applicable to <=v4.1.x.  docs.open-mpi.org is for >=v5.0.0.

Meaning: don't bother to update the old FAQ for >=v5.0.0 stuff.